var/spack/repos/builtin/packages/tau/package.py

```diff
@@ -131,6 +133,10 @@ def set_compiler_options(self, spec):
         compiler_options.append(useropt)
         return compiler_options
 
+    def setup_build_environment(self, env):
+        env.prepend_path('LIBRARY_PATH', self.spec['zlib'].prefix.lib)
+        env.prepend_path('LIBRARY_PATH', self.spec['elf'].prefix.lib)
```

Comment:
I'm surprised that these are actually needed. The compiler wrapper should add them automatically if they are link dependencies. Also, `elf` isn't necessarily in the spec, you need to check if `+libdwarf` right? Or should it be `+libelf`? The dependencies are only added for `+libdwarf`. Also, if any `elf` works, you should change the dependency above. If it has to be `libelf`, you should use `libelf` here too.

Reply:
I thought it was odd myself. This built properly in 13.3 but started failing recently in develop for extremely stripped down container builds. After a bit of hunting, the build operations spack used only worked on the command line if I did 'spack load zlib' and elf and the specific environmental setting that made the difference was LIBRARY_PATH. I would like to know the difference between setting 'depends_on' in a package file and 'spack load' in a console, but haven't tracked down that info yet.

Reply:
`spack load` sets `LIBRARY_PATH`, but Spack's build env logic clears it to prevent external packages from being sucked in.

Reply:
Is there a reason depends_on doesn't set LIBRARY_PATH with only the build/link dependencies in the build environment? Or is it supposed to?

Reply:
I'm not sure, we've never set `LIBRARY_PATH` or `CPATH` in the compiler wrappers before, we've always relied on the automatic flag injection. Maybe @tgamblin has a reason why we don't.